logstash-core/src/main/java/org/logstash/ackedqueue/Queue.java

```diff
@@ -222,17 +234,23 @@ public void open() throws IOException {
                 // but checkpoint it to update the firstUnackedPageNum if it changed
                 this.headPage.checkpoint();
             } else {
-                // head page is non-empty, transform it into a tail page and create a new empty head page
+                // head page is non-empty, transform it into a tail page
                 this.headPage.behead();
-                addPage(headCheckpoint, this.headPage);
 
-                headPageNum = headCheckpoint.getPageNum() + 1;
-                newCheckpointedHeadpage(headPageNum);
+                if (headCheckpoint.isFullyAcked()) {
+                    purgeTailPage(headCheckpoint, pageIO);
+                } else {
+                    addTailPage(headCheckpoint, this.headPage);
+                }
 
                 // track the seqNum as we add this new tail page, prevent empty tailPage with a minSeqNum of 0 to reset seqNum
                 if (headCheckpoint.maxSeqNum() > this.seqNum) {
                     this.seqNum = headCheckpoint.maxSeqNum();
                 }
+
+                // create a new empty head page
+                headPageNum = headCheckpoint.getPageNum() + 1;
+                newCheckpointedHeadpage(headPageNum);
```

Comment:
These two blocks of code swapped position, is that intentional (or doesn't matter)?

FROM
```
               headPageNum = headCheckpoint.getPageNum() + 1;
                newCheckpointedHeadpage(headPageNum);

                // track the seqNum as we add this new tail page, prevent empty tailPage with a minSeqNum of 0 to reset seqNum
                if (headCheckpoint.maxSeqNum() > this.seqNum) {
                    this.seqNum = headCheckpoint.maxSeqNum();
                }
```

TO:
```
                // track the seqNum as we add this new tail page, prevent empty tailPage with a minSeqNum of 0 to reset seqNum
                if (headCheckpoint.maxSeqNum() > this.seqNum) {
                    this.seqNum = headCheckpoint.maxSeqNum();
                }

             headPageNum = headCheckpoint.getPageNum() + 1;
              newCheckpointedHeadpage(headPageNum);
```

Reply:
yes, it does not matter - I moved the `seqNum` tracking bits for the beheaded head page right after the add/purge page handling to be consistent with the tail pages handling earlier in the code.